src/main/groovy/com/google/protobuf/gradle/ToolsLocator.groovy

```diff
@@ -112,13 +109,6 @@ class ToolsLocator {
       ext:extension ?: 'exe',
     ]
     Dependency dep = project.dependencies.add(config.name, notation)
-    FileCollection artifactFiles = config.fileCollection(dep)
-
-    for (GenerateProtoTask protoTask in protoTasks) {
-      if (protoc.is(locator) || protoTask.hasPlugin(locator.name)) {
-        protoTask.locatorToAlternativePathsMapping.put(locator.name, artifactFiles)
-        protoTask.locatorToDependencyMapping.put(locator.name, "$groupId:$artifact:$version".toString())
-      }
-    }
+    locator.resolve(config.fileCollection(dep), "$groupId:$artifact:$version".toString())
```

Comment:
Are we sure we're not resolving configuration in the configuration phase with this code?

Reply:
Do you mean: does this change _start_ resolving configuration in configuration phase? This change doesn't impact when operations are performed.

I wouldn't expect it resolves configuration, as fileCollection is lazy.